### ISSUE TITLE
Is there a way to make the bot repost only one post

### ISSUE THREAD

**Issue by Jessomadic** [MysteryPancake/Discord-Reposter#6]
I have the bot all set up but the issue is  that I would like to make it so that it only reposts the previous post

IE: Hi How is everyone doing today
/repost to Channel Hello

Then it and only it shows up in "Channel Hello". As it stands right now it just copies the whole channel that it's in and pastes it in another. But we have many channels set up based on what should go where in an Archive kind of fashion and this would be a fantastic time saver for us

**Comment by Jessomadic** [comment]
Ok so I have been able to cut down the amount of post from the whole channel to just one. by removing async function sendMessages(messages, channel, webhook, author) {
	if (inactive(channel.id)) return;
	let last;
	if (messages && messages.size) {
		const backward = messages.array().reverse();
		for (let i = 0; i < backward.length; i++) {
			if (inactive(channel.id, backward[i].channel.id)) break;
			await sendMessage(backward[i], channel, webhook, last ? last.author.id : author);
			last = backward[i];
		}
	}
}
The problem is the one it post is the first in the server vs the most recent....

**Comment by MysteryPancake** [comment]
Do you only want to repost the one message before the previous, or do you want to repost every message as new ones come as well? The live option should repost any new messages from one channel to another as long as the bot is online. I never thought of adding an option for just one message, but I just want to make sure this is what you are trying to do

**Comment by Jessomadic** [comment]
While you are at it getting rid of the info dump would work best for us as well
![image](https://user-images.githubusercontent.com/35648348/47996046-3a887280-e0c5-11e8-8801-40f2ce9bc212.png)

**Comment by MysteryPancake** [comment]
I think I understand now. Sorry about the delay between my comments, it will probably take a while for me to update this bot as I'm working on other stuff at the moment. In the meantime, if you want to get rid of the annoying info dump, then it should go away [if you remove these lines](https://github.com/MysteryPancake/Discord-Reposter/blob/master/reposter.js#L292-L349). There may also be bots offering this functionality already

**Comment by MysteryPancake** [comment]
Sorry to bother you again, I know this is probably annoying, but could you elaborate on this librarian sort of feature?

It seems that you are looking to fetch the first message before the one in which you type the channel to repost to. It seems strange if the only message you are reposting is already there. It would probably take far less time to manually select and copy the message contents rather than type a painfully long command.

I am also a bit unsure of what you mean by keywords. Initially I thought you meant the bot would sort through messages as they are posted, checking if the contents contain a string, and if so, reposting them in a specific channel. However, when re-reading, I am not sure if this is what you mean. I think you may want it to check the contents of the previous message and then repost it to the appropriate channel instead, with only that one message? This way the channel name would not need to be typed.

Again, sorry to be annoying without adding anything, but this detail will help when I get round to it.

Also, until I add it, I think you could repost only one message by [modifying these lines](https://github.com/MysteryPancake/Discord-Reposter/blob/master/reposter.js#L361-L370).
I think it may repost one message if it was modified to this:
```
const messages = await from.fetchMessages({ limit: 1, before: from.id }).catch(async function() {
	await to.send("**Can't read messages!**").catch(console.error);
});
const first = messages && messages.first();
if (first) {
	await sendMessage(first, to, webhook);
} else {
	await to.send("**Repost Complete!**").catch(console.error);
}
```
This is untested, but I hope it will work.

**Comment by Jessomadic** [comment]
No! You are being very helpful! I should have been more precise in my explanation. When I said, librarian, I was referring to the part of their job where they have to put items (books, movies, games, etc) Back where they belong. So if we are in the General Chat and someone posts a photo that should go in a NSFW channel with a #NSFW have the bot see that tag and put it in the #NSFW channel. To Archive it if you will. But We don't want it to do it to everything in the channel only the post that has the command that activates the bot (In this case being the #NSFW tag). So stating this you were correct in your initial thought of what I was trying to say! Thank you for taking your time to work on this for us! And don't worry you are not annoying at all. Ps. Here is a video of the changes not working :(  https://www.youtube.com/watch?v=FnjXKvnXI1Q&feature=youtu.be

**Comment by MysteryPancake** [comment]
I think I understand now - the tags are something like trigger words. Also, I think the change may not be working because in the video, line 305 includes `after: "0"` in fetchMessages. If this is included, Discord.js tries to get the message after message 0, which will be the first one in the channel.

**Comment by Jessomadic** [comment]
Yes, you are correct in the statement about Trigger words. Let me test this other fix for it. What would be the number I should Type for that 0 then?

**Comment by MysteryPancake** [comment]
Removing it completely should retrieve the latest message (if I remember correctly)
`const messages = await from.fetchMessages({ limit: 2 }).catch(async function() {`

**Comment by Jessomadic** [comment]
Removing that line makes the script fail to run

**Comment by Jessomadic** [comment]
New error. The bot runs but after using a command it is unable to read the messages 
![image](https://user-images.githubusercontent.com/35648348/48065548-0b433580-e199-11e8-8369-dbfba1f07434.png)

**Comment by Jessomadic** [comment]
for now it works if just the `"after: "0""` part is removed from the 
`const messages = await from.fetchMessages({ limit: 2, after: "0" }).catch(async function() {`

**Comment by Jessomadic** [comment]
Any update on this?

**Comment by MysteryPancake** [comment]
Sorry, I didn't see this comment until today. Does the bot have view message history permission? This only happens if the bot is unable to read the first message, so by removing `after: "0"`, it will be getting the latest message. If it does not error when retrieving the latest message, then it probably does not have permission to read previous messages.

Also, try changing [lines 361-363](https://github.com/MysteryPancake/Discord-Reposter/blob/master/reposter.js#L361-L363) to this:
```
const messages = await from.fetchMessages({ limit: 1, after: "0" }).catch(console.error);
```
This should give a more descriptive error in the console.
